_includes/header.html

```diff
@@ -1,8 +1,50 @@
 <header class="site-header">
+  <script type="text/javascript">
+    // By calling this on DOM Ready rather than window.load means we should have our 
+    // preferred color and classes set before the user sees the page (no flash)
+    document.addEventListener('DOMContentLoaded', () => {
+      detectColorScheme();
+    })
 
+    // This is just in case the above doesn't register in time
+    window.addEventListener('load', (e) => {
+      detectColorScheme();
+    });
+
+    function applyTheme() {
+      const checkbox = window.document.getElementById("toggleTheme");
+      window.document.getElementsByTagName('html')[0].className = window.document.getElementById('toggleTheme').checked ? "dark" : "default";
+      localStorage.setItem('theme', window.document.getElementsByTagName('html')[0].className);
+    }
+
+    //determines if the user has a set theme
+    function detectColorScheme() {
+      var theme = "light";    //default to light
+
+      //local storage is used to override OS theme settings
+      if (localStorage.getItem("theme")) {
+        if (localStorage.getItem("theme") == "dark") {
+          var theme = "dark";
```

Comment:
I'm not too used to var anymore (I tend to use [let](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/let) and [const](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/const)), I was a bit confused about the var redeclaration but it seems like it works in javascript :D

I think we can declare theme like `let theme = "light";` and write `theme = "dark";` where that makes sense, but since it works already, it s up to you :)

Reply:
Haha yeah sorrry about that it was copied from somewhere else and I didn’t change that but, will sort out